core/src/style/style.cpp

```diff
@@ -68,57 +75,104 @@ void Style::setLightingType(LightingType _type){
+        while (it != m_layers.end() && (*it)->getName() != layer.name) { ++it; }
         if (it == m_layers.end()) { continue; }
 
         // Loop over all features
         for (auto& feature : layer.features) {
 
-            /*
-             * TODO: do filter evaluation for each feature for sublayer!
-             *     construct a unique ID for a the set of filters matched
-             *     use this ID pass to the style's parseStyleParams method to construct styleParam cache
-             *     NOTE: for the time being use layerName as ID for cache
-             */
-
-            feature.props.numericProps["zoom"] = _tile.getID().z;
-
-            switch (feature.geometryType) {
-                case GeometryType::points:
-                    // Build points
-                    for (auto& point : feature.points) {
-                        buildPoint(point, parseStyleParams(it->first, it->second), feature.props, *mesh);
-                    }
-                    break;
-                case GeometryType::lines:
-                    // Build lines
-                    for (auto& line : feature.lines) {
-                        buildLine(line, parseStyleParams(it->first, it->second), feature.props, *mesh);
-                    }
-                    break;
-                case GeometryType::polygons:
-                    // Build polygons
-                    for (auto& polygon : feature.polygons) {
-                        buildPolygon(polygon, parseStyleParams(it->first, it->second), feature.props, *mesh);
-                    }
-                    break;
-                default:
-                    break;
+            std::bitset<MAX_LAYERS> uniqueID(0);
+            StyleParamMap styleParamMapMix;
+            applyLayerFiltering(feature, ctx, uniqueID, styleParamMapMix, (*it));
+
+            if(uniqueID.any()) { // if a layer matched then uniqueID should be > 0
+                feature.props.numericProps["zoom"] = _tile.getID().z;
+
+                switch (feature.geometryType) {
+                    case GeometryType::points:
+                        // Build points
+                        for (auto& point : feature.points) {
+                            buildPoint(point, parseStyleParams(styleParamMapMix), feature.props, *mesh);
+                        }
+                        break;
+                    case GeometryType::lines:
+                        // Build lines
+                        for (auto& line : feature.lines) {
+                            buildLine(line, parseStyleParams(styleParamMapMix), feature.props, *mesh);
+                        }
+                        break;
+                    case GeometryType::polygons:
+                        // Build polygons
+                        for (auto& polygon : feature.polygons) {
+                            buildPolygon(polygon, parseStyleParams(styleParamMapMix), feature.props, *mesh);
+                        }
```

Comment:
Where does the result of parseStyleParams go? And why is it computed for every feature?